validator/client/propose.go

```diff
@@ -382,9 +384,24 @@ func signVoluntaryExit(
 	return sig.Marshal(), nil
 }
 
-// Gets the graffiti from cli or file for the validator public key.
-func (v *validator) getGraffiti(ctx context.Context, pubKey [fieldparams.BLSPubkeyLength]byte) ([]byte, error) {
-	// When specified, default graffiti from the command line takes the first priority.
+// GetGraffiti gets the graffiti from cli or file for the validator public key.
+func (v *validator) GetGraffiti(ctx context.Context, pubKey [fieldparams.BLSPubkeyLength]byte) ([]byte, error) {
+	// Check proposer settings first
+	if v.proposerSettings != nil {
+		if v.proposerSettings.ProposeConfig != nil {
+			option, ok := v.proposerSettings.ProposeConfig[pubKey]
+			if ok && option.Graffiti != "" {
```

Comment:
Why checking if `option.Graffiti != ""`?
The user may set an empty graffiti for this precise validator on purpose.